### PR TITLE
Remove gulp and ts-node dev dependencies

### PR DESCRIPTION
**Pull request by awharn**
Removed the dependency on gulp, all gulp plug-ins, and ts-node, which was only used by gulp.
Removed 355 dev dependencies.
Removed cleanTypedoc functionality, which looked for a link that is no longer present in typedoc and replaced it.
Removed doc script, which generated CLIReadme, which has not been updated in 16 months and was replaced with the web-help generator and PDF.
Removed the out of date CLIReadme document.
Migrated the update licenses functionality to a plain JavaScript script.

Signed-off-by: Andrew W. Harn <andrew.harn@broadcom.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,9 +30,7 @@
     "watch": "lerna run --parallel watch",
     "watch:exe": "cd zowex && cargo install cargo-watch && cargo watch -x build -x clippy -x test",
     "watch:test": "jest --watch",
-    "doc:clean": "rimraf docs/CLIReadme.md",
-    "doc:generate": "npm run doc:clean && gulp doc",
-    "generateCleanTypedoc": "npm run typedoc && gulp cleanTypedoc",
+    "generateCleanTypedoc": "npm run typedoc",
```
Comment (zFernand0) [inline package.json:33]: should we remove this script too since it's basically just calling typedoc 😋

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

File: package.json (hunk 1 above)
Comment (awharn) [reply package.json:33]: I didn't want to just in case it was being called somewhere, but we can if you prefer.

**Review by t1m0thyj — APPROVED** [review]
🦀 Overrides be gone 🦀

**Review by gejohnston — APPROVED** [review]

**Comment by t1m0thyj** [comment]
Should we update the contributing guidelines to remove mention of gulp?
https://github.com/zowe/zowe-cli/blob/763da03c29457c4f642ed88b814da753b60c88c0/CONTRIBUTING.md?plain=1#L182

**Comment by awharn** [comment]
> Should we update the contributing guidelines to remove mention of gulp?
> 
> https://github.com/zowe/zowe-cli/blob/763da03c29457c4f642ed88b814da753b60c88c0/CONTRIBUTING.md?plain=1#L182

I think we can do that, and I have done so on my local branch. Is there anything else that should be inserted instead? Or should we just have `Use build tasks to enforce rules where possible.` on the `Build Process Guidelines` section?

**Comment by t1m0thyj** [comment]
> I think we can do that, and I have done so on my local branch. Is there anything else that should be inserted instead? Or should we just have `Use build tasks to enforce rules where possible.` on the `Build Process Guidelines` section?

I'd be ok with removing the line. Not sure what to insert in place of gulp besides "We use npm scripts", which sounds kind of obvious.

**Pull request merged**